railties/test/generators/argv_scrubber_test.rb

```diff
@@ -56,6 +57,21 @@ def self.default_rc_file
         assert_equal [], args
       end
 
+      def test_xdg_config_no_custom_rc
+        xdg = Dir.tmpdir
+        ENV["XDG_CONFIG_HOME"] = xdg
+        rails_xdg_path = File.join(xdg, "rails")
+        Dir.mkdir(rails_xdg_path, 0700) unless Dir.exist?(rails_xdg_path)
+        Dir.new(rails_xdg_path)
+        railrc_path = File.join(rails_xdg_path, "railsrc")
+        railsrc = File.new(railrc_path, "w")
+        railsrc.close
+
+        scrubber = Class.new(ARGVScrubber)
+        assert_equal scrubber.default_rc_file, railrc_path
+        File.delete(railrc_path) if File.exist?(railrc_path)
```

Comment:
Let's move to an `ensure` block and also unset the `ENV["XDG_CONFIG_HOME"]` value.

Reply:
Thanks @rafaelfranca, I actually ended up switching to Jonathan Hefners code below, but good point on unsetting the ENV, I hadn't considered that.